File: src/main/java/testCodi/a03PermMissingElem/solution_perm_missing_element.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution_perm_missing_element(A):
    # write your code in Python 3.6
    # zero or one elements
    if len(A) == 0:
        return 1
    A.sort()
    # the first item
    if A[0] != 1:
        return 1
    # the last item
    if A[len(A) - 1] != len(A) + 1:
        return len(A) + 1
    for index in range(1, len(A) + 1):
        if index < A[index - 1]:
            return A[index - 1] - 1
    return 0

File: src/main/java/testCodi/a03PermMissingElem/solution_perm_missing_element.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution_perm_missing_element(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    if (A.length == 0) {
        return 1;
    }
    A = A.sort();
    if (A[0] != 1) {
        return 1;
    }
    if (A[A.length - 1] != A.length + 1) {
        return A.length + 1;
    }
    for (var index = 1; index <= A.length; index++) {
        if (index < A[index - 1]) {
            return A[index - 1] - 1;
        }
    }
    return 0;
}